pkg/controller/kfdef/kfdef_controller.go

```diff
@@ -437,3 +494,100 @@ func setAnnotations(configPath string, annotations map[string]string) error {
 	config.SetAnnotations(anns)
 	return kfloaders.WriteConfigToFile(*config)
 }
+
+// getClusterServiceVersion retries the clusterserviceversions available in the operator namespace.
+func getClusterServiceVersion(cfg *rest.Config, watchNameSpace string) (*olm.ClusterServiceVersion, error) {
+
+	operatorClient, err := olmclientset.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("error getting operator client %v", err)
+	}
+	csvs, err := operatorClient.ClusterServiceVersions(watchNameSpace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	// get csv with CR KfDef
+	if len(csvs.Items) != 0 {
+		for _, csv := range csvs.Items {
+			for _, operatorCR := range csv.Spec.CustomResourceDefinitions.Owned {
+				if operatorCR.Kind == string(kftypesv3.KFDEF) {
+					return &csv, nil
+				}
+			}
+		}
+	}
+	return nil, nil
+}
+
+// operatorUninstall deletes all the externally generated resources. This includes monitoring resources and applications
+// installed by KfDef.
+func operatorUninstall(c client.Client, cfg *rest.Config) error {
+	// Get watchNamespace
+	operatorNamespace, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		return err
+	}
+
+	// Delete generated namespaces
+	generatedNamespaces := &v1.NamespaceList{}
+	nsOptions := []client.ListOption{
+		client.MatchingLabels{rhodsGeneratedNamespaceLabel: "true"},
```

Comment:
I'm not seeing this label anywhere on the projects/namespaces....for the redhat-ods-monitoring namespace, I see the following labels...
    hive.openshift.io/managed: "true"
    olm.operatorgroup.uid/b4e6a5a4-779a-4145-a6fc-403343705480: ""
    openshift.io/cluster-monitoring: "true"

Reply:
Oh wait...I see that the deployer is expected to add those labels...I do not have that running with that change yet.

Reply:
The change to add labels recently merged in odd-deployer : https://github.com/red-hat-data-services/odh-deployer/pull/84 

Did you pull in latest commit?

Reply:
Yes...I see now.  Totally just me confusing myself.

Reply:
It should not be prefixed `rhods`